lib/memcache/connection.ex

```diff
@@ -71,7 +75,10 @@ defmodule Memcache.Connection do
   ]
 
   defp with_defaults(opts) do
+    default_timeout = Application.get_env(:memcachex, :default_timeout, :infinity)
```

Comment:
I prefer not to touch application env especially if the application is not singleton. None of the other configs are controlled via application env as well. I would suggest to add it to the `@default_opts`

Reply:
Hm, yes, I recognize that this was the only place using the Application env. Given that the `@default_opts` are hardcoded, though, is your preference then that there is no way to globally configure the timeout, rather that a timeout should be passed in the opts with every call?

Reply:
> Rather, that a timeout should be passed in the opts with every call (that wishes to not use infinity)?

It's only required for start_link on Memcache/Connection module. So I expect it to be needed at one place (wherever the memcache supervisor hierarchy is configured)

Reply:
Alright, makes sense. I'll put together some changes early next week, and ping you when it's ready to be looked at again. Thanks for the quick review!